src/NzbDrone.Core/Tv/AddSeriesService.cs

```diff
@@ -0,0 +1,103 @@
+        private readonly IAddSeriesValidator _addSeriesValidator;
+        private readonly Logger _logger;
+
+        public AddSeriesService(ISeriesService seriesService,
+                                IProvideSeriesInfo seriesInfo,
+                                IBuildFileNames fileNameBuilder,
+                                IAddSeriesValidator addSeriesValidator,
+                                Logger logger)
+        {
+            _seriesService = seriesService;
+            _seriesInfo = seriesInfo;
+            _fileNameBuilder = fileNameBuilder;
+            _addSeriesValidator = addSeriesValidator;
+            _logger = logger;
+        }
+
+        public Series AddSeries(Series newSeries)
+        {
+            Ensure.That(newSeries, () => newSeries).IsNotNull();
+
+            newSeries = SeriesGetSeriesInfo(newSeries);
+
+            if (string.IsNullOrWhiteSpace(newSeries.Path))
+            {
+                var folderName = _fileNameBuilder.GetSeriesFolder(newSeries);
+                newSeries.Path = Path.Combine(newSeries.RootFolderPath, folderName);
+            }
+
+            var validationResult =_addSeriesValidator.Validate(newSeries);
+
+            if (!validationResult.IsValid)
+            {
+                throw new ValidationException(validationResult.Errors);
+            }
+
+            _logger.Info("Adding Series {0} Path: [{1}]", newSeries, newSeries.Path);
+
+            newSeries.CleanTitle = newSeries.Title.CleanSeriesTitle();
+            newSeries.SortTitle = SeriesTitleNormalizer.Normalize(newSeries.Title, newSeries.TvdbId);
+            newSeries.Added = DateTime.UtcNow;
+
+            _seriesService.AddSeries(newSeries);
+
+            return newSeries;
+        }
+
+        private Series SeriesGetSeriesInfo(Series newSeries)
+        {
+            Tuple<Series, List<Episode>> tuple;
+
+            try
+            {
+                tuple = _seriesInfo.GetSeriesInfo(newSeries.TvdbId);
+            }
+            catch (SeriesNotFoundException)
+            {
+                _logger.Error("tvdbid {1} was not found, it may have been removed from TheTVDB.", newSeries.TvdbId);
+                
+                throw new ValidationException(new List<ValidationFailure>
+                                              {
+                                                  new ValidationFailure("TvdbId", "A series with this ID was not found", newSeries.TvdbId)
+                                              });
+            }
+
+            var series = tuple.Item1;
+
+            series.Path = newSeries.Path;
+            series.RootFolderPath = newSeries.RootFolderPath;
+            series.Monitored = newSeries.Monitored;
+            series.AddOptions = newSeries.AddOptions;
+            series.ProfileId = newSeries.ProfileId;
+            series.SeasonFolder = newSeries.SeasonFolder;
+            series.SeriesType = newSeries.SeriesType;
```

Comment:
Do all these properties have sensible defaults? Imagine someone adding a series without `{ ..., "monitored": true }`.

Also, it might be better to extract setting these properties to a separate method, or even class.
iirc we have some code that defines what the user is allowed to change on a series. It's a powerplay in SeriesModule between seriesService.UpdateService and first having to fetch the current state from the seriesService. Basically, SeriesResource.ToModel only applies properties the user is allowed to change. (https://github.com/Sonarr/Sonarr/blob/816cf608fcef07e375aa750a863d0a7365108981/src/NzbDrone.Api/Series/SeriesResource.cs#L208-L225)
So having the same list of 'mutable' properties on two locations isn't very maintainable.